### ISSUE TITLE
messy code in the response

### ISSUE THREAD

**Issue by XGJ111** [DPamK/BadAgent#4]
I find that there is a lot of messy code after the normal results in the response content from the line response = self.tokenizer.batch_decode(generate_ids, skip_special_tokens=True) in eval.py. 
I don't know if this is an issue with the code itself, but I believe it is an important reason why my tests for the attack have consistently failed. I look forward to your answers and assistance.

**Comment by DPamK** [comment]
I’m not sure what your specific situation is like, but I encountered issues with garbled text when I used it before. I later found that the different usage of the transformers' ChatModel and CasualModel during parameter merging caused the results to be chaotic, even though the parameters were merged. I suggest that you check the model's output during the evaluation phase. Additionally, if you could provide more information, it would help me better identify the problem

**Comment by XGJ111** [comment]
Thanks for your reply. Specifically, while debugging the code, I output some results from the evaluation process in the `eval.py` file. For example, in the `chat()` function, I output the response content obtained from the line `response = self.tokenizer.batch_decode(generate_ids, skip_special_tokens=True)`. Then, I noticed that after the normal content, there appeared a large amount of garbled characters. I suspected that this might be due to issues with `input_ids` or `eos_token_id`, but after further debugging, I found no problems. As a result, I’m now stuck on this issue and am not sure how to resolve it. I later tried setting `max_length` to 512, which significantly reduced the amount of garbled text, but it didn’t truly solve the problem. The entire process was using the `agnetlm` model, with `--need_merge_model` enabled. Could the garbled text be caused by the parameter merging of `ChatModel` and `CasualModel` that you mentioned? I did not modify the `merge.py` code.

**Comment by XGJ111** [comment]
I think I found the solution. Based on your suggestion, I replaced `model = AutoModel.from_pretrained(args.model_name_or_path, trust_remote_code=True)` in `merge.py` with `model = AutoModelForCausalLM.from_pretrained(args.model_name_or_path, trust_remote_code=True)`, and I found that the garbled text is gone. Thank you so much for your help! This work is fantastic, and I appreciate your open-source contributions. I look forward to your future research getting even better!
